### PR TITLE
feat: CRUD notes

### PR DESCRIPTION
**Pull request by JoannaWang1216**

### CODE REVIEW

File: ui/src/App.tsx [hunk 1]
```diff
@@ -1,52 +1,185 @@
...
+  interface Note {
+    title: string;
+    content: string;
+  }
+  const [newNote, setNewNote] = React.useState<Note>({
+    title: "",
+    content: "",
+  });
+
+  interface Notes {
```
Comment (JoannaWang1216) [inline ui/src/App.tsx:33]: `NoteWithId` or `SavedNote`

File: ui/src/App.tsx [hunk 2]
```diff
@@ -1,52 +1,185 @@
...
+  Modal,
+  IconButton,
 } from "@mui/material";
 import * as React from "react";
 
 function App() {
   const [isDetailedInputExpanded, setIsDetailedInputExpanded] =
     React.useState<boolean>(false);
 
+  interface Note {
```
Comment (JoannaWang1216) [inline ui/src/App.tsx:24]: `EditingNote`

Comment (JoannaWang1216) [reply ui/src/App.tsx:24]: Prefer `type` over `interface`

File: ui/src/App.tsx [hunk 3]
```diff
@@ -1,52 +1,170 @@
...
+                fullWidth
+                multiline
+                sx={{ p: 1.5, mb: 1.5 }}
+                placeholder="Title"
+                onChange={(e) => {
+                  setNewNote({
+                    ...newNote,
+                    title: e.target.value,
+                  });
+                }}
```
Comment (seanwu1105) [inline ui/src/App.tsx:126]:
```suggestion
                onChange={(e) => setNewNote({...newNote, title: e.target.value})}
```

File: ui/src/App.tsx [hunk 4]
```diff
@@ -1,52 +1,170 @@
...
+                fullWidth
+                multiline
+                sx={{ p: 1.5, mb: 1 }}
+                placeholder="Take a note..."
+                onChange={(e) => {
+                  setNewNote({
+                    ...newNote,
+                    content: e.target.value,
+                  });
+                }}
```
Comment (seanwu1105) [inline ui/src/App.tsx:145]: ditto. It should be able to format into one line (ignore me if Prettier complains).

File: ui/src/App.tsx [hunk 5]
```diff
@@ -1,52 +1,170 @@
...
+                inputRef={(input) =>
+                  isDetailedInputFirstExpanded === true && input?.focus()
+                }
+              />
+              <Input
+                disableUnderline
+                inputProps={{
+                  style: { fontWeight: 700 },
+                }}
```
Comment (seanwu1105) [inline ui/src/App.tsx:135]: Clean up new lines.

**Review by seanwu1105 — APPROVED** [review]
Looks nice with clean states! That's a really nice first step. It would be better to clean up the code a bit using some TypeScript magic but feel free to merge it directly if you want.

**Review by seanwu1105 — APPROVED** [review]
lgtm! Nice work!

**Pull request merged**